File: bridge/ChannelSplitterNode.cpp
#include "dart_api/dart_api.h"
#include "LabSound/LabSound.h"
#include "KeepNode.cpp"
#include "struct.h"
using namespace lab;

DART_EXPORT int createChannelSplitterNode(AudioContext* context) {
    auto node = std::make_shared<ChannelSplitterNode>(*context);
    return keepNode(node);
}

DART_EXPORT void ChannelSplitterNode_addOutputs(int nodeId, int n) {
    auto node = std::static_pointer_cast<ChannelSplitterNode>(getNode(nodeId));
    if(node) node->addOutputs(n);
}

File: lab_sound_ffi/lib/src/lab_sound/channel_splitter_node.dart
import 'audio_context.dart';
import 'audio_node.dart';
import 'lab_sound.dart';

class ChannelSplitterNode extends AudioNode {
  ChannelSplitterNode(AudioContext ctx, {int numberOfOutputs = 1}): super(ctx, LabSound().createChannelSplitterNode(ctx.pointer, numberOfOutputs));
  addInputs(int n) => LabSound().ChannelSplitterNode_addOutputs(nodeId, n);
}